app-misc/pip3line/pip3line-3.4.2.ebuild

```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6,7}} )
+
+inherit cmake-utils python-r1
+
+DESCRIPTION="Raw bytes manipulation, transformations (decoding and more) and interception"
+HOMEPAGE="https://github.com/metrodango/pip3line"
+
+KEYWORDS=""
+SRC_URI=""
+
+if [[ ${PV} == 9999* ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/metrodango/pip3line.git"
+	EGIT_BRANCH="master"
+else
+	SRC_URI="https://github.com/metrodango/pip3line/archive/v${PV}.tar.gz  -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="GPL-3"
+SLOT="0"
+
+IUSE="distorm openssl qscintilla"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-qt/qtconcurrent:5
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxmlpatterns:5
+	distorm? ( dev-libs/distorm64 )
+	openssl? ( dev-libs/openssl:0= )
+	qscintilla? ( x11-libs/qscintilla )"
+
+DEPEND="${RDEPEND}
+	distorm? ( dev-vcs/git )"
+
+src_configure() {
+	if [[ "${PYTHON_TARGETS}" == *python3* ]]; then
+		local python3_usage=yes
+	else
+		local python3_usage=no
+	fi
+
+	local mycmakeargs=(
+		-DBASIC=yes
+		-DWITH_OPENSSL=$(usex openssl)
+		-DWITH_PYTHON27=$(usex python_targets_python2_7)
+		-DWITH_PYTHON3=${python3_usage}
```

Comment:
you do it wrong, I was telling you about  python targets based mycmakeargs changes, not about 3rd party variables

Reply:
I am going to move away from using python-r1 entirely. Please see current discussion with @hsoft .

Reply:
I think we misunderstood each other. I don't think you should move away from `python-r1`. We still need to target proper python impls with `python_targets_*`. I meant that you would need to add additional `python/python2` use flags to enable/disable python3/python2 support and then use `REQUIRED_USE` to ensure that only one impl of each class is selected.

Reply:
ok then. But I am not sure how to implement this because the cmake scripts currently only take the highest version available.

Reply:
Welcome to the world of packaging. Usually, when upstream's build system is broken or insufficiently flexible, we patch our way through it, but you're upstream, so you can fix your own build system.

Reply:
I agree and this is already (moslty) included because one can pass some values for the Python includes files and lib files.

My problem was more about how to pass the correct values based on the python_targets* selected via portage.

Reply:
should be fixed now